.circleci/config.yml

```
@@ -0,0 +1,49 @@
+#
+# Circle CI configuration. Runs each time we push a new commit to Github.
+#
+version: 2.1
+
+jobs:
+  build_latest_ocaml:
+    docker:
+      - image: ocaml/opam:ubuntu-20.04-ocaml-4.13
+    working_directory: ~/atd
+    steps:
+      - checkout
+      - run:
+          name: Install system dependencies
+          command: ./.circleci/setup-system
+      - run:
+          name: Install opam dependencies
+          command: ./.circleci/setup-opam
+      - run:
+          name: Build
+          command: opam exec -- make
+      - run:
+          name: Test
+          command: opam exec -- make test
+  build_min_ocaml:
+    docker:
+      - image: ocaml/opam:ubuntu-20.04-ocaml-4.08
+    working_directory: ~/atd
+    steps:
+      - checkout
+      - run:
+          name: Install system dependencies
+          command: ./.circleci/setup-system
+      - run:
+          name: Install opam dependencies
+          command: ./.circleci/setup-opam
+      - run:
+          name: Build
+          command: opam exec -- make
+      - run:
+          name: Test
+          command: opam exec -- make test
```

I think that you can remove the duplication. Circle ci supports matrix (probably the nicest solution) and commands (easiest one). I don't know the exact syntax though

An example with commands
```suggestion
commands:
  build:
    steps:
      - checkout
      - run:
          name: Install system dependencies
          command: ./.circleci/setup-system
      - run:
          name: Install opam dependencies
          command: ./.circleci/setup-opam
      - run:
          name: Build
          command: opam exec -- make
      - run:
          name: Test
          command: opam exec -- make test

jobs:
  build_latest_ocaml:
    docker:
      - image: ocaml/opam:ubuntu-20.04-ocaml-4.13
    working_directory: ~/atd
    steps:
      - build:

  build_min_ocaml:
    docker:
      - image: ocaml/opam:ubuntu-20.04-ocaml-4.08
    working_directory: ~/atd
    steps:
      - build:
```

---

nice. I see the matrix feature is [demonstrated here](https://circleci.com/blog/circleci-matrix-jobs/), but let's use your solution.